xdsl/dialects/stencil.py

```diff
@@ -747,21 +756,24 @@ def get(res: Sequence[SSAValue | Operation]):
         return ReturnOp.build(operands=[list(res)])
 
     def verify_(self) -> None:
+        unroll_factor = self.unroll_factor
         types = [
             o.type.elem if isinstance(o.type, ResultType) else o.type for o in self.arg
         ]
         apply = cast(ApplyOp, self.parent_op())
         res_types = [cast(TempType[Attribute], r.type).element_type for r in apply.res]
-        if len(types) != len(res_types):
+        if len(types) != len(res_types) * unroll_factor:
             raise VerifyException(
-                f"stencil.return expected {len(res_types)} operands to match the parent "
+                f"stencil.return expected {len(res_types) * unroll_factor} operands to match the parent "
                 f"stencil.apply result types, got {len(types)}"
             )
-        if types != res_types:
-            raise VerifyException(
-                "stencil.return expected operand types to match the parent "
-                "stencil.apply result element types."
-            )
+        for i, res_type in enumerate(res_types):
+            for type in types[unroll_factor * i : unroll_factor * (i + 1)]:
+                if type != res_type:
+                    raise VerifyException(
+                        "stencil.return expected operand types to match the parent "
+                        "stencil.apply result element types."
+                    )
```

Comment:
might be worth mentioning `i` here? or would that be confusing client-side?

Reply:
Updated, WDYT?

Reply:
Much clearer, thank you